opensquirrel/circuit.py

```diff
@@ -101,9 +101,10 @@ def merge_single_qubit_gates(self) -> None:
         """Merge all consecutive 1-qubit gates in the circuit.
```

Comment:
This method should be made into a generic `merge` method.

```
    def merge(self, merger: Merger) -> None:
        """Generic merge pass.
        It applies the given merger to the circuit.
        """
        merger.merge(self)
```

Reply:
Hi Chris, I was trying to add this step. However, as soon as I do ```from opensquirrel.passes.merger.general_merger import Merger```, I get a KeyError because there is a circular dependency (Circuit needs Merger, Merger needs Circuit). Do you know how I can fix this?

Reply:
I have attempted to just pass the class name instead of the instance:
```def merge_single_qubit_gates(self, merger: type[Merger]) -> None:```
The tests pass, but MyPy doesn't like this. I am thinking of a solution that would let the tests pass and keeps MyPy happy, but I haven't found one yet.

Reply:
Did you see my comment above?

Reply:
Hi Roberto, I didn't refresh the page so I didn't see your comment; will implement it right now :)

Reply:
Another solution is simply to move the `from opensquirrel.circuit import Circuit` in `single_qubit_gates_merger.py` after `class SingleQubitGatesMerger(Merger)`.